client/components/jetpack/threat-item/utils.ts

```diff
@@ -1,7 +1,27 @@
 import { translate } from 'i18n-calypso';
-import { Threat, ThreatFix, ThreatType } from './types';
+import { SignatureComponents, Threat, ThreatFix, ThreatType } from './types';
 import type { TranslateResult } from 'i18n-calypso';
 
+export const getThreatSignatureComponents = ( threat: Threat ): SignatureComponents | null => {
+	// ([signatureId])?[language]_[payload]_[family]_[variant]
+	const signatureRegex = /(?:\(([^)]+)\))?([^_]+)_([^_]+)_(.+)_(.+)/g;
```

Comment:
* Optional signature ID in the front i.e. `(123)foo_bar_baz_bop` or `foo_bar_baz_bop` are both valid.
* Family absorbs any extra underscores, i.e. `(123)foo_bar_baz_bop_beep` will have a `family` value of `baz_bop`.
* All components other than signature ID must exist for a match i.e. `(123)foo_bar` will return `null`.